docs/integrations/step-patterns.md

```diff
@@ -0,0 +1,326 @@
+            });
+            await jobState.addRelationship(groupUserRelationship);
+          };
+        );
+      }
+    );
+  },
+}
+```
+
+## "Fetch Child Entities" step pattern
+
+In our example, the `/devices/{deviceId}/applications` endpoint is not
+appropriate for the **Fetch Relationships** pattern since we haven't already
+ingested applications from a top-level `/applications` endpoint (the API spec
+did not define one). In this case, the **Fetch Child Entities** pattern is most
+appropriate, and will handle ingestion of both the `acme_device_application`
+entities and the `acme_device_has_application` relationships.
+
+The archetype for the **Fetch Child Entities** pattern:
+
+```typescript
+const fetchDeviceApplications: Step = {
+  id: 'fetch-device-applications',
+  name: 'Fetch Device Applications',
+  entities: [{ _type: 'acme_device_application', ... }],
+  relationships: [{ _type: 'acme_device_has_application', ... }],
+  dependsOn: ['fetch-devices'],
+  executionHandler: (context) => {
+    const { jobState, instance } = context;
+    const client = new AcmeClient({ config: instance.config });
+
+    await jobState.iterateEntities(
+      { _type: 'acme_device' },
+      async (deviceEntity) => {
+        const device = await getRawData<AcmeDevice>(deviceEntity);
+
+        await client.iterateDeviceApplications(
+          { deviceId: device.id },
+          async (application) => {
+            const applicationEntity = createApplicationEntity(application);
+            await jobState.addEntity(applicationEntity);
+
+            const deviceApplicationRelationship = createDirectRelationship({
+              from: deviceEntity,
+              _class: RelationshipClass.HAS,
+              to: applicationEntity,
+            });
+            await jobState.addRelationship(deviceApplicationRelationship);
+          };
+        );
+      }
+    );
+  },
+}
+```
+
+## "Build Child Relationships" step pattern
+
+Recall from our API spec that there is no endpoint directly linking `acme_user`s
+to their `acme_device`. Instead, the API response for the `/users` endpoint
+returns the user's device ID.
+
+```
+GET /users
+{
+  id: 'user-id',
+  device: { id: 'device-id' },
+  ...
+}
+```
+
+Here, we use the **Build Child Relationships** pattern to build these
+relationships once both the `fetch-users` and `fetch-devices` steps have
+completed. This pattern is unique in that it is the only pattern which does not
+require the `AcmeClient`.
+
+The archetype for the **Build Child Relationships** pattern:
+
+```typescript
+const buildUserDeviceRelationships: Step = {
+  id: 'build-user-device-relationships',
+  name: 'Build User to Device Relationships',
+  entities: [],
+  relationships: [{ _type: 'acme_user_has_device', ... }],
+  dependsOn: ['fetch-users', 'fetch-devices'],
+  executionHandler: (context) => {
+    const { jobState, instance } = context;
+    const client = new AcmeClient({ config: instance.config });
```

Comment:
Wouldn't we just make this relationship when fetching the device, as this client isn't being used in this example?

Reply:
Yes, great catch!